src/Button/DigitizeButton/DigitizeButton.spec.tsx

```diff
@@ -134,133 +124,132 @@ describe('<DigitizeButton />', () => {
+        wrapper.instance().componentDidMount();
+        expect(createModifyInteraction).toHaveBeenCalledTimes(1);
 
-        wrapper.setState({
-          interactions: [mockInteraction]
+        expect(wrapper.instance()._modifyInteraction).toBeInstanceOf(OlInteractionModify);
+        expect(wrapper.instance()._modifyInteraction.getActive()).toBeFalsy();
+
+        createModifyInteraction.mockRestore();
+      });
+
+      it ('creates a translate interaction on mount', () => {
+        const wrapper = setupWrapper(true);
+        wrapper.setProps({
+          drawType: null,
+          editType: 'Edit'
         });
+        const createTranslateInteraction = jest.spyOn(wrapper.instance(), 'createTranslateInteraction');
 
-        expect(map.getInteractions().getLength()).toBe(defaultMapInteractionsLength + 1);
-        // Warning: using of private properties such `listeners_` could be
-        // a bit fragile. We should probably find another way to get the
-        // appropriate value.
-        expect(map.listeners_.pointermove).toBeDefined();
+        expect(createTranslateInteraction).toHaveBeenCalledTimes(0);
 
-        wrapper.instance().onToggle(false);
+        wrapper.instance().componentDidMount();
+        expect(createTranslateInteraction).toHaveBeenCalledTimes(1);
 
-        expect(map.listeners_.pointermove).toBeUndefined();
+        expect(wrapper.instance()._translateInteraction).toBeInstanceOf(OlInteractionTranslate);
+        expect(wrapper.instance()._translateInteraction.getActive()).toBeFalsy();
+
+        createTranslateInteraction.mockRestore();
       });
 
-      // eslint-disable-next-line max-len
-      it ('unregisters `add` listener on digitize feature collection if drawType is Text and button was untoggled', () => {
+      it ('removes all interactions and map listeners from the map on unmount', () => {
         const wrapper = setupWrapper();
-        wrapper.setProps({
-          drawType: 'Text'
-        });
-        const mockInteraction = getMockDrawPointInteraction();
 
-        const instance = wrapper.instance();
-        instance.createDigitizeLayer();
-        instance._digitizeFeatures = instance._digitizeLayer.getSource().getFeaturesCollection();
+        const defaultMapInteractionsLength = map.getInteractions().getLength();
+        map.addInteraction(getMockDrawPointInteraction());
+        map.on('pointermove', wrapper.instance().onPointerMove);
 
-        map.addInteraction(mockInteraction);
-        instance._digitizeFeatures.on('add', wrapper.instance().handleTextAdding);
+        wrapper.instance().componentDidMount();
 
-        expect(instance._digitizeFeatures.listeners_.add.length).toBe(2);
+        expect(map.getInteractions().getLength()).toBe(defaultMapInteractionsLength + 1);
+        // Warning: using of private properties such `listeners_` could be
+        // a bit fragile. We should probably find another way to get the
+        // appropriate value.
```

Comment:
Since this is only in the test, this is IMO acceptable for the moment, but the comment is of course totally valid.